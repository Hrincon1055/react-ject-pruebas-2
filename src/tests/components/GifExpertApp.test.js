import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Pruebas de componente GifExpertApp.js", () => {
  const title = "titulo";
  const url = "http://sdfegr";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);
  test("Debe de mostrar componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("parra con el title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });
  test("Debe de tener una imagen", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });
  test("Debe tener clase de animacion", () => {
    const div = wrapper.find("div");
    const clasName = div.prop("className");
    expect(clasName.includes("animate__fadeIn")).toBe(true);
  });
});
