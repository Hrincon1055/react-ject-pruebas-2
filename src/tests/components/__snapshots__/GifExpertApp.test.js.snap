// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de componente GifExpertApp.js Debe de mostrar componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="titulo"
    src="http://sdfegr"
  />
  <p>
     
    titulo
     
  </p>
</div>
`;
